pycloudlib/azure/cloud.py

```diff
@@ -665,64 +798,96 @@ def launch(
                 f"{self._type} launch requires image_id param."
                 f" Found: {image_id}"
             )
+        if not ip_addresses_params:
+            ip_addresses_params = [None]
+        if not network_interfaces:
+            network_interfaces = [None]
+
+        if len(ip_addresses_params) > len(network_interfaces):
+            raise PycloudlibError(
+                "IP addresses should not be more than the NICs provided."
```

Comment:
I think with this message, we directly refer to the input parameters and could be simpler to understand for users.

```suggestion
                "The number of `ip_addresses_params` cannot be bigger than the number of `network_interfaces`."
```